### PR TITLE
OVH Cloud (Openstack)

### PR DESCRIPTION
**Pull request by xlucas**
Hello there!

I've been working during the past days on providing *an* Openstack cloud implementation (see #105). Openstack is used by many operators to provide public and private clouds.

However, there's actually no such thing as *the* Openstack implementation. Operators usually differ from one another by the collection of services they provide or not, by the way some of this components behave, and also by having different regions, different flavors, different images, different network topologies, and the list goes on.

For this reason, I thought it would be particularly well fitted to have:
- A generic set of Openstack resources (keypair, network, subnet, security groups and so on)
- Operator-bound profile and model (as well as resources if any), where specificities can be expressed.

Working myself at OVH, a large European cloud provider  which delivers a Public Cloud infrastructure based on Openstack, here's the first Openstack profile for Kubicorn !

### CODE REVIEW

**Comment by xmudrii** [comment]
@xlucas Thank you much for such a great PR! Very well done! I've left you a comment—you need to update headers for new files.

Once tests are sorted, I think it's okay to merge this.

**Comment by xlucas** [comment]
@xmudrii Actually the headers are already there. Looks like the check fails because the year is 2018 and not 2017.

**Comment by xmudrii** [comment]
@xlucas yep. maybe it doesn't make sense at all, but script assumes it's 2017. Let's stay with 2017 for now. I can check with @kris-nova can we update headers.

**Comment by xmudrii** [comment]
@xlucas Can you please rebase against the master branch? Maybe you want to look into the #546 to see more about the latest changes.

**Comment by xlucas** [comment]
@xmudrii Done

**Comment by xmudrii** [comment]
@xlucas A tiny nitpick. Personally, I don't prefer underlines in the Go file names, beside for test files, etc. For example, I prefer `securitygroup.go` over `security_group.go`.

I think it's the only occurrence. It's not a big thing and we can go without it, but maybe it's good idea to rename the file. In case for the profile file, I think it can stay, as other profiles are using it as well (sometimes, we're not consistent at all :P).

As said on Slack, if nobody takes on in the meanwhile, I'll do the final review tomorrow.

**Comment by xlucas** [comment]
Well, I did some renaming.

File: cloud/openstack/openstackSdk/sdk.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+// Sdk represents an Openstack SDK.
+type Sdk struct {
+	Compute       *gophercloud.ServiceClient
+	Network       *gophercloud.ServiceClient
+	ObjectStorage *gophercloud.ServiceClient
+}
+
+// NewSdk constructs a new Openstack SDK for the specified region.
+//
+// The following environment variable list is looked up in the user
```
Comment (xmudrii) [inline cloud/openstack/openstackSdk/sdk.go]: Can we please document this. Those variables should be added to the [`envar.md`](https://github.com/kris-nova/kubicorn/blob/master/docs/_documentation/envar.md) file.

**Comment by xmudrii** [comment]
@xlucas Sorry for the delay! I was very busy for the past few days.
Can we just document this? Beside that, this is very awesome. Thank you so much for such a great job!

**Comment by xmudrii** [comment]
:shipit: once :green_apple: :)

**Comment by xmudrii** [comment]
Thank you so much for all the great work! We really appreciate it! :shipit:

**Pull request merged**